packages/backend-common/src/ffmpeg.ts

```diff
@@ -0,0 +1,19 @@
+import { exec } from 'node:child_process';
+import { logger } from '@guardian/transcription-service-backend-common';
+import { promisify } from 'node:util';
+
+const execPromise = promisify(exec);
+
+export const getFileDuration = async (filePath: string): Promise<number> => {
+	const command = `ffprobe -i ${filePath} -show_entries format=duration -v quiet -of csv="p=0"`;
+	try {
+		const { stdout, stderr } = await execPromise(command);
+		if (stderr) {
+			logger.error(` ffprobe stderr: `, stderr);
+		}
+		return parseFloat(stdout);
+	} catch (error) {
+		logger.error(`Error during ffprobe file duration detection`, error);
+		throw error;
```

Comment:
will we return a 500 if we can't find file duration here? might be better to default to whisperx

Reply:
good shout, implemented in https://github.com/guardian/transcription-service/pull/136/commits/5d7709a195510dd82070c4b929c33d9578d7fa95